Dockerfile

```diff
@@ -1,20 +1,28 @@
-FROM golang:1.19.5 AS operator-build
-
-ARG GO_LINKER_ARGS
-
-COPY . /app
+# setup build image
+FROM golang:1.19.5 AS go-base
+RUN \
+    --mount=type=cache,target=/var/cache/apt \
+    apt-get update && apt-get install -y libbtrfs-dev libdevmapper-dev
+
+# download go dependencies
+FROM go-base AS go-mod
 WORKDIR /app
+COPY go.mod go.sum ./
+RUN go mod download && go mod verify
 
-RUN apt-get update && \
-    apt-get install -y libbtrfs-dev libdevmapper-dev
-
+# build operator binary
+FROM go-mod AS operator-build
+ARG GO_LINKER_ARGS
+COPY src ./src
```

Comment:
we don't copy the `third_party_licenses` folder anymore, which is created as part of `push_image.sh`.

We can't drop these licenses, so it should be installed and cached as part of the `Dockerfile` 🙂